app/main.py

```diff
@@ -1,2 +1,157 @@
+from typing import Any, Hashable, NamedTuple
+
+
+DELETED = object()
+
+
+class Pair(NamedTuple):
+    key: Any
+    value: Any
+
+    def __str__(self) -> str:
+        return f"{self.key!r}: {self.value!r}"
+
+    def __repr__(self) -> str:
+        return f"({self.key!r}, {self.value!r})"
+
+
 class Dictionary:
-    pass
+    @classmethod
+    def from_dict(cls, dictionary: object, capacity: int = None) -> Any:
+        hash_table = cls(capacity or len(dictionary))
+        for key, value in dictionary.items():
+            hash_table[key] = value
+        return hash_table
+
+    def __init__(self,
+                 capacity: int = 8,
+                 load_factor_threshold: float = 0.6) -> None:
+        if capacity < 1:
+            raise ValueError("Capacity must be a positive")
+        if not (0 < load_factor_threshold <= 1):
+            raise ValueError("Load factor must be a number between (0, 1]")
+        self._slots = capacity * [None]
+        self._load_factor_threshold = load_factor_threshold
+
+    def __len__(self) -> int:
+        return len(self.pairs)
+
+    def __iter__(self) -> Any:
+        yield from self.keys
+
+    def __delitem__(self, key: Hashable) -> None:
+        for index, pair in self._probe(key):
+            if pair is None:
+                raise KeyError(key)
+            if pair is DELETED:
+                continue
```

Comment:
Why if is deleted do continue?
![Uploading image.png…]()

Reply:
I can't set to none. In case:
When I have collision, and I put into next slot my pair, and after I delete previous. If I set to none, I can't find my pair in next slots.

Reply:
But in this case better also raise exception
<img width="763" alt="image" src="https://github.com/mate-academy/py-dictionary/assets/65166203/638a1b31-1de5-4007-a577-770dd8ed8850">

Reply:
But okay for now. The rest code is okay)